recipe/meta.yaml

```diff
@@ -78,6 +78,8 @@ outputs:
       run_constrained:
         - ld64 {{ ld64_version }}.*
         - cctools {{ cctools_version }}.*
+        # clang might pull in the wrong cctools otherwise
+        - libllvm{{ llvm_version }}
```

Comment:
Note that this doesn't exclude `libllvm{{other version}}` because multiple libllvm major versions don't conflict with each other. What it seems to need is a build variant pin on the `cctools` package (or another way to _exclude_ other libllvm major versions). So you can pick `libllvm14` here, and still install `cctools` with `libllvm15`